File: C++/W3Schools/Functions_Tutorial/5_Recursion/5_Recursion.cpp

// C++: W3Schools - Main Tutorial
// Section 5: Recursion

#include <iostream>
using namespace std;

// BASICS
// Functions can call other functions. Recursion is when a function calls itself:
int sum(int k) {
  if (k > 0) {
    return k + sum(k - 1);
  } else {
    return 0;
  }
}

int main() {
  int result = sum(10);
  cout << "1. " << result << '\n';
  return 0;
}

File: W3Schools/Backend/Java/Functions_Tutorial/5_Recursion.java

// Java: W3Schools - Functions Tutorial
// Section 5: Recursion

public class Main {

  public static void main(String[] args) {
    int result = sum(10);
    System.out.println(result);
  }

  // A function can call itself. This is known as recursion:
  public static int sum(int k) {
    if (k > 0) {
      return k + sum(k - 1);
    } else {
      return 0;
    }
  }
}
